src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -31,13 +31,13 @@ public void add(T value, int index) {
                 valueToAdd = new Node<>(tail, value, null);
                 valueToAdd.prev.next = valueToAdd;
                 tail = valueToAdd;
-            } else if (nodeInIndex.prev == null && size > 0) {
-                valueToAdd = new Node<>(null, value, nodeInIndex);
-                nodeInIndex.prev = valueToAdd;
-                head = valueToAdd;
             } else {
-                valueToAdd = new Node<>(findNodeByIndex(index).prev, value, findNodeByIndex(index));
-                valueToAdd.prev.next = valueToAdd;
+                valueToAdd = new Node<>(nodeInIndex.prev, value, nodeInIndex);
+                if (nodeInIndex.prev == null && size > 0) {
```

Comment:
```suggestion
                if (nodeInIndex.prev == null) {
```
we have checked `size == 0` and it's false, and size cannot be negative.